jupyterhub/templates/proxy/autohttps/deployment.yaml

```diff
@@ -41,90 +29,66 @@ spec:
       terminationGracePeriodSeconds: 60
       nodeSelector: {{ toJson .Values.proxy.nodeSelector }}
       {{- include "jupyterhub.coreAffinity" . | nindent 6 }}
+      volumes:
+        - name: certificates
+          emptyDir: {}
+        - name: traefik-config
+          configMap:
+            name: traefik-proxy-config
+      initContainers:
+      - name: load-acme
+        image: "{{ .Values.proxy.secretSync.image.name }}:{{ .Values.proxy.secretSync.image.tag }}"
+        {{- with .Values.proxy.secretSync.image.pullPolicy }}
+        imagePullPolicy: {{ . }}
+        {{- end }}
+        command: ["/usr/local/bin/secret-sync.py", "load", "proxy-public-tls-acme", "acme.json", "/etc/acme/acme.json"]
+        env:
+        # We need this to get logs immediately
+        - name: PYTHONUNBUFFERED
+          value: "True"
+        volumeMounts:
+          - name: certificates
+            mountPath: /etc/acme
       containers:
-        - name: nginx
-          image: "{{ .Values.proxy.nginx.image.name }}:{{ .Values.proxy.nginx.image.tag }}"
-          {{- with .Values.proxy.nginx.image.pullPolicy }}
+        - name: traefik
+          image: "{{ .Values.proxy.traefik.image.name }}:{{ .Values.proxy.traefik.image.tag }}"
+          {{- with .Values.proxy.traefik.image.pullPolicy }}
           imagePullPolicy: {{ . }}
           {{- end }}
           resources:
-            {{- .Values.proxy.nginx.resources | toYaml | trimSuffix "\n" | nindent 12 }}
-          args:
-            - /nginx-ingress-controller
-            - --default-backend-service={{ .Release.Namespace }}/proxy-http
-            - --configmap={{ .Release.Namespace }}/nginx-proxy-config
-            - --ingress-class=jupyterhub-proxy-tls
-            - --watch-namespace={{ .Release.Namespace }}
-            {{- if .Values.debug.enabled }}
-            - --v=3
-            {{- end }}
-          env:
-            - name: POD_NAME
-              valueFrom:
-                fieldRef:
-                  fieldPath: metadata.name
-            - name: POD_NAMESPACE
-              valueFrom:
-                fieldRef:
-                  fieldPath: metadata.namespace
-          livenessProbe:
-            httpGet:
-              path: /healthz
-              port: 10254
-              scheme: HTTP
-            initialDelaySeconds: 10
-            timeoutSeconds: 1
-          readinessProbe:
```

Comment:
We could use the /ping endpoint here. Since it is a deployment now that should be able to do rolling updates, we benefit slightly from it. Example:

If we start up the new traefik pod during a rolling update, the old pod will start terminating as soon as the traefik container has started running even though it hasn't become ready. If we have a readiness check k8s wont shut down the previous traefik instance until it has become ready for real and for example able to respond to pings.

Reply:
This should be a HTTP endpoint I think, because if not, we will never be able to meet an initial http01 challange.

Reply:
I created a new issue to represent this work to be done in a future PR instead.

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1546

Reply:
Thank you!